docs/contributing/database-migrations/edd.mdx

```diff
@@ -173,49 +222,82 @@ END
 </TabItem>
 </Tabs>
 
-## Workflow
+## Our EDD Process
+
+There are some unique constraints to Bitwarden that are not addressed directly in Martin Fowler's
+EDD article.
+
+- Our Production instances in the cloud are required to be on at all times
+- Our self-host instances that we do not directly have access to manage must support the same EDD
+  processes; however, they do not have the same always-on application constraint
+- Minimization of manual steps in our process
+
+The process to support all of these constraints is a complex one. Below is an image of a state
+machine that will hopefully help visualize the process and what it supports. It assumes that all
+database changes follow the standards that are laid out in [Migrations](./).
 
-The Bitwarden specific workflow for writing migrations are described below.
+---
 
-### Developer
+![Bitwarden EDD State Machine](./edd_state_machine.jpg) \[Open Image in a new tab for better
+viewing\]
 
-The development flow is described in [Migrations](./).
+---
 
-### Devops
+### Cloud Environments
 
-#### On `rc` cut
+Since we treat both schema migrations and data migrations as just migrations, the only issues that
+we are solving for is orchestrating the runtime constraints on the migration. Eventually, all
+migrations will end up in `DbScripts`. However, to control the timing of running _Transition_ and
+associated _Finalization_ migrations, we need to keep them outside of `DbScripts` until the correct
+timing.
 
-Create a PR moving the future scripts.
+In our environments with always-on applications, _Transition_ scripts must be run after the new code
+has been rolled out. To execute a full deploy, we run all new migrations in `DbScripts`, roll out
+the new code, and then run all _Transition_ migrations in the `DbScripts_transition` directory as
+soon as all of the new code services are online. In the case of a critical failure after the new
+code is rolled out, we will conduct a Rollback (see Rollbacks below). _Finalization_ migrations will
+not be run until the start of the next deploy when they are moved into `DbScripts`.
 
-- `DbScripts_future` to `DbScripts`, prefix the script with the current date, but retain the
-  existing date.
-- `dbo_future` to `dbo`.
-  <bitwarden>
-    <li>
-      Create a ticket in Jira with a `Due Date` of the release date to ensure future migrations are
-      merged in and ready to be executed. Set the ticket that created the future migration as a
-      blocker.
-    </li>
-  </bitwarden>
+After this deploy, to prep for the next release, all migrations in `DbScripts_transition` are moved
+to `DbScripts` and then all migrations in `DbScripts_finalization` are moved to `DbScripts`,
+conserving their execution order for a clean install. For our current branching strategy, PRs will
+be open against `master` when `rc` is cut to prep for this release. This PR automation will also
+handle renaming the migration file and updating any reference of `[dbo_future]` to `[dbo]`.
```

Comment:
⚠️ This "future" naming may need revision if you want your naming changes to carry into here, but technically I suppose it's fine to leave if we want to maintain.

Reply:
Done